### ISSUE TITLE
api.yml menu is hardcoded

### ISSUE THREAD

**Issue by Fryguy** [ManageIQ/manageiq.org#814]
The api.yml does drives the left side menu, but it is hardcoded, and the actual api docs love over in manageiq_docs.  As such, changes are required in 2 repos.  Instead we should probably autogenerate at build time.

cc @apinnick

**Comment by Fryguy** [comment]
cc @abellotti @bdunne

**Comment by chessbyte** [comment]
@Fryguy are you referring to this [file](https://github.com/ManageIQ/manageiq.org/blob/master/site/_data/menus/api_menu.yml)?

**Comment by Fryguy** [comment]
@chessbyte Yes.  Anytime a new section is added or removed over in manageiq_docs for the API, someone has to remember to come here and also add it to the yml.  Feels like we need a way to autogenerate that from the content in manageiq_docs instead.

**Comment by Fryguy** [comment]
I've been thinking about this and I think the better approach is to have the menu in the documentation repo itself (menu file per branch)...Then it can be added as part of the PR that introduces the doc.  I don't think we should autogenerate, because then we can't control the order of documents.  However, we _may_ want to introduce a test/warning if a document is created but not in the menu.  This way the PR author can choose to add it to the menu or put it in a "known hidden" list.
